labs/b4.md

```diff
@@ -0,0 +1,269 @@
+copies files into the correct location, and updates the OS registry to reflect
+this.
+
+Enter the package manager, a centralized way to install, update, and remove
+software from your computer using verified sources called repositories.  Most
+package managers work in the same way: the package manager gets a list of
+packages from the repository, and then, when asked to install a package,
+fetches the package from the repository, verifies that it is legit, and
+installs it. A package manager is like a librarian. When a patron wants to read
+a book, the librarian consults a catalog (which is the package database) and
+then fetches the book from the shelf and gives it to the patron (installing the
+package).  Sometimes, the librarian has to update the catalog because new
+editions of the books were added (updating the package database). When the
+patron is done, the librarian knows where to return the book (cleaning up after
+removing a package).  Contrast this to the approach on Windows, where dedicated
+uninstaller programs are occasionally necessary as the OS does not know where
+programs install their own files.
+
+## Debian: An Example
+
+In this class, we will be focused on using Debian. As noted before, Debian uses
+`apt` or `dpkg` as its package manager.
+
+We invoke the package manager by using `apt`. Before installing anything, you
+shoud update your local copy of packages and versions to ensure the package
+manager installs the latest and most up-to-date packages. To do that, run:
+
+    $ sudo apt update
+
+To find the package to install:
+
+    $ apt search [package|description]
+
+To install a package, run:
+
+    $ sudo apt install package
+
+To remove a package, run:
+
+    $ sudo apt remove package
+
+Easy? When you want to upgrade the packages that you have installed when
+new versions are released, you can do so by calling:
+
+    $ sudo apt upgrade
+
+There are also other commands, such as removing unneeded dependencies (`sudo
+apt autoremove`) and purging packages (`sudo apt purge`) but that is what the
+man pages are for.
+
+### Example
+
+We are going to install GCC for the next step of the lab.
+
+**This, and all other parts of the lab, should be done on your DigitalOcean
+VMs, as you cannot install packages on OCF machines.**
+
+Please note that you can only connect to your VM from inside the Berkeley
+network, so you will have to either be on campus wifi, or you will have to SSH
+through `ssh.ocf.berkeley.edu` (or a similar on-campus host) to access your VM.
+You should be able to connect to it at `<OCF username>.decal.xcf.sh` by running
+`ssh <username>@<username>.decal.xcf.sh` and the password you set when your
+first logged in, or using the email you received.
+
+Simply run:
+
+    $ sudo apt install gcc
+
+Now check if GCC is installed by running the following to check GCC's version:
+
+    $ gcc --version
+
+Now install the `ruby-dev` and `ruby-ffi` packages for the next parts of the
```

Comment:
Explain what these packages are, or show them how to use `apt show $pkg` to learn. (pref both)